### DISCUSSION TITLE
sync icons on file explorer

### DISCUSSION THREAD

**Discussion by pkwizard** [winfsp/winfsp#589] [Ideas]
I want to create a user experience similar to what you see in services like OneDrive or Dropbox, where a green checkmark is displayed next to a directory or file once an upload is successfully completed. This should apply not only to newly uploaded entities but also to any files or directories that have been downloaded. In addition, a sync icon (such as a spinning circle or progress indicator) should appear while the file is in the process of uploading or downloading from remote, providing clear feedback to the user about the current sync status.

Can this be achieved in some way?

**Comment by fsgeek** [comment]
It’s done via a shell extension with an icon overlay. It has nothing to do
with the file system (though the shell extension can query the file system
to figure out the file state.)
